spring-cloud-gcp-data-datastore/src/test/java/org/springframework/cloud/gcp/data/datastore/it/TestEntityRepository.java

```diff
@@ -33,12 +35,16 @@
 	@Query(value = "select size from  test_entities_ci where size <= @size", count = true)
 	int countEntitiesWithCustomQuery(@Param("size") long size);
 
-	@Query(value = "select * from  test_entities_ci where id = @id_val", exists = true)
+	@Query(value = "select __key__ from  test_entities_ci "
+			+ "where id = @id_val", exists = true, keyQuery = true)
 	boolean existsByEntitiesWithCustomQuery(@Param("id_val") long id);
 
 	@Query("select id from  test_entities_ci where id <= @id_val ")
 	List<TestEntity> findEntitiesWithCustomProjectionQuery(@Param("id_val") long id);
 
+	@Query(value = "select __key__ from test_entities_ci", keyQuery = true)
```

Comment:
Why can't we determine that it's a key query based on the return type? I believe Mark mentioned that it's possible.

Reply:
Though I think we can do this another way. i'll add a commit shortly.

Reply:
@meltsufin PTAL. I removed this setting but had to do it a different way.

Reply:
The issue before the most recent commit @meltsufin was that we didn't know if we should try to map the result or return it directly. However by losing some type-safety we can treat entity queries and key queries together. It is a tradeoff for type safety, but I think worth it.

Reply:
Thanks! I think this is on the right track now.
What would happen if you allowed a Key query to go through the mapping machinery? Also, what if it's not a Key query, but something like a custom projection, think about the selecting plates from cars table and return `List<String>`?

Reply:
- So right now we can tell if the query is a Key or not based on the results of the query and not the return type of the function. I think this is safest because there are no assumptions being made. If the query was a projection or full row query (it's the same to us), then we will map the objects and the query method will return those. At this point there will be a cast exception because the entities cannot be cast to Key (at runtime if the query method is called). 

- Regarding `List<String>`. There are 2 possibilities. If the user had given a Key query, then we will return the list of keys. However due to type erasure I don't think this will be a runtime cast error unless the user actually tries to access an object inside the list at which point the Key to String cast will fail.  On the other hand if the user had given a projection or full entity query then he will get back a List of POJOs and the same situation would happen (cast error when accessing an item). 

We can tackle supporting the general `List<SimpleType>` issue separately in the next PR and it wouldn't conflict with this PR.

Reply:
My understanding is that we have access to both the method return type and query result set type. So, it should simply be a matter of running the result set through the conversion machinery with the conversion target being the return type of the method. What am I missing here? @dmitry-s can you please weigh in on this?